Sources/NostrSDK/CustomEmoji.swift

```diff
@@ -39,6 +39,14 @@ public class CustomEmoji: CustomEmojiValidating, Equatable {
     }
 }
 
+public protocol CustomEmojiBuilding: NostrEventBuilding {}
+public extension CustomEmojiBuilding {
+    func customEmojis(_ customEmojis: [CustomEmoji]) -> Self {
```

Comment:
This example illustrates why using a builder pattern is powerful for something like Nostr. There are all kinds of shared event creation patterns across different Nostr event kinds. Each kind-specific `NostrEvent.Builder` subclass can easily integrate with the protocol and get the implementation for free. In the case for custom emojis, it can be used across [three different event kinds](https://github.com/nostr-protocol/nips/blob/master/30.md) -- I've only integrated with two of them (kind 0 metadata and kind 1 text notes).